classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -32,752 +34,1144 @@ abstract protected function getContextTable(): string;
+            // Custom field (not found in at least one of the softwares)
+            $affectedRows += $this->deleteRequiredReference('submissions', 'context_id', $this->getContextTable(), $this->getContextKeyField());
+
+            // Attempts to recover the field submissions.current_publication_id before discarding the entry
+            $rows = DB::table('submissions AS s')
+                ->leftJoin('publications AS p', 'p.publication_id', '=', 's.current_publication_id')
+                ->join(
+                    'publications AS last',
+                    fn(JoinClause $q) => $q->where(
+                        fn(Builder $q) => $q->from('publications AS p2')
+                            ->whereColumn('p2.submission_id', '=', 's.submission_id')
+                            ->orderByDesc('p2.publication_id')
+                            ->limit(1)
+                            ->select('p2.publication_id'),
+                        '=',
+                        DB::raw('last.publication_id')
+                    )
+                )
+                ->whereNull('p.publication_id')
+                ->pluck('s.submission_id', 'last.publication_id');
+            foreach ($rows as $publicationId => $submissionId) {
+                $this->_installer->log("The current publication ID ({$publicationId}) for the submission ID {$submissionId} is invalid, the publication ID {$publicationId} will replace it");
+                $affectedRows += DB::table('submissions')->where('submission_id', '=', $submissionId)->update(['current_publication_id' => $publicationId]);
             }
 
+            // The current_publication_id is nullable, but it's in fact required by the software, so we delete orphan entries instead of nulling them
+            $affectedRows += $this->deleteRequiredReference('submissions', 'current_publication_id', 'publications', 'publication_id');
+            return $affectedRows;
+        });
+
+        $this->addTableProcessor('submission_files', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~5 entities: file_id->files.file_id genre_id->genres.genre_id source_submission_file_id->submission_files.submission_file_id submission_id->submissions.submission_id uploader_user_id->users.user_id
+            $affectedRows += $this->deleteRequiredReference('submission_files', 'submission_id', 'submissions', 'submission_id');
+            $affectedRows += $this->deleteRequiredReference('submission_files', 'file_id', 'files', 'file_id');
+            $affectedRows += $this->cleanOptionalReference('submission_files', 'uploader_user_id', 'users', 'user_id');
+            $affectedRows += $this->cleanOptionalReference('submission_files', 'source_submission_file_id', 'submission_files', 'submission_file_id');
+            $affectedRows += $this->cleanOptionalReference('submission_files', 'genre_id', 'genres', 'genre_id');
```

Comment:
The `genre_id` is always filled, perhaps it's worth to grab a fallback value here too? I did a small test and it worked fine with null.